src/pages/index.js

```diff
@@ -25,28 +36,135 @@ const cardData = {
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api.getUserInfo().then((userData) => {
+  //console.log("Setting avatar to:", userData.avatar);
+  //console.log("User Data from API:", userData);
+  userInfo.setUserInfo({
+    name: userData.name,
+    job: userData.about,
+    avatar: userData.avatar,
+  });
+});
+
+/* ------------------------------------------------- */
+/*                     Form Validation
+/* ------------------------------------------------- */
+const formValidators = {};
+
+const enableValidation = (settings) => {
+  const formList = Array.from(document.querySelectorAll(settings.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(settings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(settings);
+
 // /* ------------------------------------------------- */
-// /*                 Profile Edit Form
+// /*                 Profile Avatar
 /* ------------------------------------------------- */
+const handleAvatarFormSubmit = (formData) => {
+  const newAvatarUrl = formData.avatarUrl;
+
+  return api
+    .updateAvatar(newAvatarUrl)
+    .then((userData) => {
+      console.log("Avatar updated on server:", userData.avatar);
+      userInfo.setUserAvatar(userData.avatar);
+      profileEditAvatar.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    });
+};
+
+const profileEditAvatar = new PopupWithForm(
+  "#update-avatar-modal",
+  handleAvatarFormSubmit,
+  (formValues) => {
+    profileEditAvatar.renderLoading(true);
+    const avatarUrl = formValues.avatar;
+    api
+      .updateAvatar(avatarUrl)
+      .then((userData) => {
+        console.log("Avatar URL:", userData.avatar);
+        userInfo.setUserAvatar(userData.avatar);
+        profileEditAvatar.close();
+        validator.resetValidation();
```

Comment:
`        validator.resetValidation();
`

Please, delete it

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card or change the avatar, you don't have any errors after submitting. You need to call `disableButton` there.